index.js

```diff
@@ -38,36 +29,20 @@ function fastifyStatic (fastify, opts, next) {
     lastModified: opts.lastModified,
     maxAge: opts.maxAge
   }
-  const page500 = opts.page500Path || DEFAULT_500_PAGE
-  const page403 = opts.page403Path || DEFAULT_403_PAGE
-  const page404 = opts.page404Path || DEFAULT_404_PAGE
-
-  function overwriteStatusCode (res, statusCode) {
-    return function () { res.statusCode = statusCode }
-  }
-
-  function servePathWithStatusCodeWrapper (page, statusCode) {
-    return function servePage (req, res) {
-      send(req, page)
-        .on('stream', overwriteStatusCode(res, statusCode))
-        .pipe(res)
-    }
-  }
-  const serve404 = servePathWithStatusCodeWrapper(page404, 404)
-  const serve403 = servePathWithStatusCodeWrapper(page403, 403)
-  const serve500 = servePathWithStatusCodeWrapper(page500, 500)
 
-  function pumpSendToReply (req, reply, pathname) {
-    const sendStream = send(req, pathname, sendOptions)
+  function pumpSendToReply (request, reply, pathname) {
+    const sendStream = send(request.req, pathname, sendOptions)
 
     if (setHeaders !== undefined) {
       sendStream.on('headers', setHeaders)
     }
 
     sendStream.on('error', function (err) {
-      if (err.statusCode === 404) return serve404(req, reply.res)
-      if (err.statusCode === 403) return serve403(req, reply.res)
-      serve500(req, reply.res)
+      if (err.status === 404) {
+        fastify._404Context.handler(request, reply)
```

Comment:
I’m not a huge fan of accessing a private propery. I think we should expose the handler through a documented method.

Reply:
Sure. What do you think that should look like? `fastify.callNotFoundHandler(request, reply)`? Or maybe add it to the `reply` object as `reply.notFound()`?

Reply:
I prefer reply.notFound()

Reply:
Okay 👍 I'll propose this in Fastify with a PR